js/lex.go

```diff
@@ -161,6 +161,9 @@ func (l *Lexer) Next() (TokenType, []byte) {
 	// ErrorToken, WhitespaceToken and CommentToken are already returned
 	if tt == LineTerminatorToken || tt == PunctuatorToken && regexpStateByte[c] {
 		l.regexpState = true
+	} else if tt == IdentifierToken {
+		hash := ToHash(l.r.Lexeme())
+		l.regexpState = hash != 0 && hash != This
```

Comment:
Could work, that's kinda what we did in Acorn with https://github.com/ternjs/acorn/blob/master/src/tokentype.js#L112-L146, I'm just slightly worried about complexity / performance downside of using a separate hash table (given that `switch` is usually optimized better even when results in jump table too).

What other strings that are not keywords do you think can be added to JS hashes in future?